modular_skyrat/modules/modular_items/lewd_items/code/lewd_organs/_genital.dm

```diff
@@ -1,4 +1,31 @@
 /obj/item/organ/external/genital
-	var/datum/reagents/internal_fluids
-	var/list/contained_item
-	var/obj/item/inserted_item //Used for toys
+	/// The fluid count of the genital.
+	var/internal_fluid_count = 0
+
+	/// The maximum amount of fluid that can be stored in the genital.
+	var/internal_fluid_maximum = 0
+
+	/// The datum to be used for the tracked fluid, should it need to be added to a fluid container.
+	var/internal_fluid_datum
+
+	/// The currently inserted sex toy.
+	var/obj/item/inserted_item
+
+/// Helper proc for checking if internal fluids are full or not.
+/obj/item/organ/external/genital/proc/internal_fluid_full()
+	return internal_fluid_count >= internal_fluid_maximum
+
+/// Adds the given amount to the internal fluid count, clamping it between 0 and internal_fluid_maximum.
+/obj/item/organ/external/genital/proc/adjust_internal_fluid(amount)
+	amount += internal_fluid_count
+	internal_fluid_count = clamp(amount, 0, internal_fluid_maximum)
+
+/// Tries to add the specified amount to the target reagent container. Keeps in mind internal_fluid_count.
+/obj/item/organ/external/genital/proc/transfer_internal_fluid(datum/reagents/reagent_container, attempt_amount)
```

Comment:
This one should probably return how much it was able to transfer, maybe? Not sure if it's really necessary right now, but if there's a check that tells you how much you transferred, might as well make it say the real value.

Reply:
I did think about doing that too. I'll do it for future's sake if wanted, though I highly doubt anyone will make the ERP system that advanced without maintainers getting mad about performance and maintainability.

Reply:
Fair enough.